_posts/2023-06-26-angular-standalone-components.md

```diff
@@ -0,0 +1,181 @@
+
+4. Delete the `NgModule` and the routing configurations.
+
+{% include
+components/figure.html
+url="/assets/images/post/angular-standalone-components-3.png"
+description="Migrate a component to a Standalone"
+%}
+
+# Differences between using NgModules and Standalone feature
+
+## Bootstrapping an Application
+
+Once the `AppComponent` is standalone you can get rid of the `AppModules` completely and use  
+[bootstrapApplication](https://angular.io/api/platform-browser/bootstrapApplication "https://angular.io/api/platform-browser/bootstrapApplication") API to bootstrap the application in `src/main.ts` file and importing the dependencies directly in the `AppComponent`, you can find more about this [here](https://angular.io/guide/standalone-components#bootstrapping-an-application-using-a-standalone-component "https://angular.io/guide/standalone-components#bootstrapping-an-application-using-a-standalone-component").
+
+{% include
+components/figure.html
+url="/assets/images/post/angular-standalone-components-1.png"
+description="Bootstrapping an Application"
+%}
+
+## Changes in the Routing configuration
+
+With standalone feature in place, `NgModule` is no long required for many lazy loading scenarios, you can simply lazy load a standalone component by exporting the routes and using `loadComponent` instead of `loadChildren` and then provide them in the `src/main.ts` files using `provideRouter()` from `@angular/router` as shown below.
+
+{% include
+components/figure.html
+url="/assets/images/post/angular-standalone-components-2.png"
+description="Changes in the Routing configuration"
+%}
+
+## Specs for Standalone component
+
+The TestBed configuration now doesn't needs to import all the dependencies used by the component  
+you are testing and hence reduces the boilerplate code in the spec file.
+
+{% include
+components/figure.html
+url="/assets/images/post/angular-standalone-components-4.png"
+description="Specs for Standalone component"
+%}
+
+_**NOTE:**_ After the migration the specs might be broken and might need manual fixes
+
+# Improvements on performance using standalone feature
+
+Here are a few ways in which Angular standalone components can contribute to performance improvements:
+
+* **Reduced bundle size:** You can reduce the overall bundle size of your app by using Angular's standalone feature and eliminating NgModules and the boilerplate code that comes with it.
+
+* **Isolation and lazy loading:** You can enhance the performance by encapsulating functionality within standalone components which helps to load all the resource only when it is lazy loaded.
+
+* **Reusability:** Angular standalone components are self-contained, independent, and can be re-utilized across the application.
```

Comment:
Is there any change in how to package and distribute a standalone component compared to the NgModule ones?

Reply:
@tcorral There is no change in packaging and distribution with standalone components.
This point highlights that standalone components are self-contained and could be imported directly into other components without having to import the dependencies again and again like we use to do with NgModules